### PR TITLE
feat: cURL import now happens via a modal

### LINKED ISSUE #32942: [Task]: Import cURL action doesn't adhere to the interaction paradigm of IDE 2.0. It needs to show up in an overlay.
[The current behaviour of the `import cURL` action breaks the list-detail pattern](https://pasteboard.co/kmQdlOuUQMXq.png)

The import cURL option does not create a list item. It is instead an _**action**_ which takes in a cURL command and upon successful import gets converted to an API. 

Actions should be rendered in a modal and the solution here is to show the import cURL action also in a modal. Upon successful import, a list-detail view of an API is created.

### PR DESCRIPTION
**Pull request by hetunandu**
## Description

Adds the cURL import flow to go through a modal instead of editor level screen. This is done to ensure the IA is correct as per IDE 2.0 standards

Fixes #32942

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!CAUTION]
> 🔴 🔴 🔴 Some tests have failed.
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9869174972>
> Commit: 6143c1c79c0bb7c6a4899fc477985d0ed23699df
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9869174972&attempt=2&selectiontype=test&testsstatus=failed&specsstatus=fail" target="_blank">Cypress dashboard</a>.
> Tags: @tag.All
> Spec: 
> The following are new failures, please fix them before merging the PR: <ol>
> <li>cypress/e2e/Regression/ClientSide/OtherUIFeatures/Omnibar_spec.js
> <li>cypress/e2e/Regression/ClientSide/PartialImportExport/PartialExport_spec.ts</ol>
> <a href="https://internal.appsmith.com/app/cypress-dashboard/identified-flaky-tests-65890b3c81d7400d08fa9ee3?branch=master" target="_blank">List of identified flaky tests</a>.
> <hr>Wed, 10 Jul 2024 07:56:03 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **Refactor**
  - Updated test case function calls and refactored imports to improve code structure and maintainability in multiple files related to CURL import flows.
  - Consolidated and rearranged steps in CURL import flow test cases for better readability and organization.

- **New Features**
  - Introduced `AppCURLImportModal` component to handle CURL command imports within the API editor.
  - Added constants `SET_CURL_MODAL_OPEN` and `SET_CURL_MODAL_CLOSE` for managing the state of the CURL modal.

- **Style**
  - Updated selector formats and added proper commas in the `ApiEditor.js` file.

- **Chores**
  - Removed unused constants and functions related to CURL import page paths across multiple files for cleaner codebase.
  - Streamlined the management of the CURL import modal state in the reducer and added relevant selectors.

### CODE REVIEW

File: app/client/src/pages/Editor/IntegrationEditor/AIDataSources.tsx [hunk 1]
```diff
@@ -105,10 +105,6 @@ export const CardContentWrapper = styled.div`
 `;
 
 interface Props {
-  history: {
```
Comment (albinAppsmith) [inline app/client/src/pages/Editor/IntegrationEditor/AIDataSources.tsx:108]: Is this relevant in this PR?

**Review by albinAppsmith — DISMISSED** [review]

**Comment by albinAppsmith** [comment]
Saw this behaviour in DP and not sure we should be handling this. When I manually enter cURL URL, the page becomes empty without even and invalid URL error.
Screenshot 2024-07-10 at 13 36 13

**Comment by hetunandu** [comment]
> @hetunandu Saw this behaviour in DP and not sure we should be handling this. When I manually enter cURL URL, the page becomes empty without even and invalid URL error. Screenshot 2024-07-10 at 13 36 13

I dont think its a problem for curl as this behaviour is present for all non standard urls. Maybe we can do something at an IDE level to show such errors

Created an issue for this: https://github.com/appsmithorg/appsmith/issues/34846

**Pull request merged**